src/test/java/org/dcsa/bkg/model/transferobjects/DocumentPartyTOTest.java

```diff
@@ -63,6 +63,25 @@ void testToVerifyPartyFunctionIsRequired() {
         violations.stream().anyMatch(v -> "PartyFunction is required.".equals(v.getMessage())));
   }
 
+  @Test
+  @DisplayName(
+      "DocumentPartyTO should throw error if partyFunction is set to a value that is not in its enum subset.")
+  void testToVerifyNotAllowedEnumsInPartyFunctionIsInvalid() {
+
+    for (String str : new String[] {"AG", "VSL", "ATH", "PLT", "TR", "TWG", "BUK", "LSH"}) {
```

Comment:
Is this a good way to do this test? What happens when the enum is expanded with new values - these new values will not be caught by this test? (Or am I misreading this?)

Reply:
@HenrikHL Updated to test all values except for the enumSubset. If the subset changes then the test case should fail so it doesn't go unnoticed and we can update it from there.